### PR TITLE
Fix MDN URLs in api folder

### LINKED ISSUE #5201: Linter for mdn_url
Sometimes the `__compat.mdn_url` property is incorrect, e.g. as shown in #5195. After seeing that issue, I looked for more similar bugs and found a couple, so I propose we make an explicit linter check for `mdn_url`. Below is a draft checklist, please tell me if it sounds reasonable.

# Checks
1. Any top-level feature has a proper link, e.g. `api.feature` has a link `https://developer.mozilla.org/docs/Web/API/feature`
2. Any sub-feature has a link that extends from the parent
E.g. `api.feature.attribute` has a link `https://developer.mozilla.org/docs/Web/API/feature/attribute` and `api.feature.feature_event` has link `https://developer.mozilla.org/docs/Web/API/feature/feature_event`

### PR DESCRIPTION
**Pull request by bershanskiy**
# Summary
Fix MDN URLs in `api` folder.
Currently, `mdn_url`s are full of mischief like [linking `Magnetometer.x` to `Magnetometer.y`](https://github.com/mdn/browser-compat-data/blob/ae92ae9c611aad41cd59571b46cf80cbe11efcad/api/Magnetometer.json#L101). :)

This linter finds all URLs that don't match either of two patterns:
```
https://developer.mozilla.org/docs/Web/API/feature/subfeature
https://developer.mozilla.org/docs/Web/API/feature#subfeature
```
There are still a few non-standard URLs, which I think should just be whitelisted individually. (Details below.)

1. `CSSMathNegate.values` was renamed into `CSSMathNegate.value` because that is the [standard name](https://drafts.css-houdini.org/css-typed-om-1/#cssmathnegate).
2. `Headers.headers` renamed to `Headers.Headers` because it is a constructor.
3. `MediaKeyStatusMap.iterator` renamed to `MediaKeyStatusMap.@@iterator` to match existing naming convention.

# Related
#5201 is a tracking bug for all MDN URLs.
#5195 was the original issue that demonstrated existence of incorrect URLs.

A checklist to help your pull request get merged faster:
- [x] Summarize your changes
- [ ] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)
- [ ] Data: if you tested something, describe how you tested with details like browser and version
- [x] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [x] Link to related issues or pull requests, if any

### CODE REVIEW

**Comment by vinyldarkscratch** [comment]
Hey @bershanskiy, thanks for taking initiative on this one!  I agree, this is a big problem that we should have resolved.  A couple of things, though:

1) Can we split off all of the changes into a separate PR?  This will help us during the review process.
2) Moreso, I think this should follow the bulk update process we've defined, since this will be such a large number of updates.
3) I think this can apply for more than just the API data; CSS, JavaScript, HTML, etc. features may also experience this issue.  Can we add support for other folders as well, and make this a per-file test, rather than testing the entire folder?  (This will also fix an issue I noticed where this linter file checks the entire API folder **every** file, instead of just once per run of `npm test`.)

**Comment by bershanskiy** [comment]
> Can we split off all of the changes into a separate PR? This will help us during the review process.

Do you mean split the data changes into a separate PR?

> Moreso, I think this should follow the bulk update process we've defined, since this will be such a large number of updates.

I'm not sure it's possible to make a script to update these files because there are too many edge cases. E.g., when to use slash and when to use `#`? This depends mostly on MDN article structure, so a script would need to fetch external resource and make a choice based on it -- not very elegant. Also, some URLs reference section titles in English article, so a script will need to fetch the article in a specific locale (English) and check for a specific section existence -- pretty ugly as well.

> I think this can apply for more than just the API data

This is true. I started with API data just to try out the method. I agree: the whole data should be linted, ideally.

> This will also fix an issue I noticed where this linter file checks the entire API folder every file, instead of just once per run of npm test.)

Ooops... Good catch!

Another thing I noticed is: many `description`s (e.g., for events) are linted to have a very specific format suitable for English locals only. If the format is known in advance, why bother saving it in the data in the first place? Why not just provide a function that creates these descriptions and, may be, insert them into release "builds" of the data sent to `npm`?

**Comment by vinyldarkscratch** [comment]
> Do you mean split the data changes into a separate PR?

Ack, forgot to say "data" -- that's what I mean, yeah!

> I'm not sure it's possible to make a script to update these files because there are too many edge cases.

Hmm, yeah, you do have a point...  This makes me wonder if we should have a bulk update process for changes that are better to be performed manually, rather than by a script -- maybe we should just keep it broken up into small PR chunks?

> Another thing I noticed is: many `description`s (e.g., for events) are linted to have a very specific format suitable for English locals only. If the format is known in advance, why bother saving it in the data in the first place? Why not just provide a function that creates these descriptions and, may be, insert them into release "builds" of the data sent to `npm`?

This is a really good point, and one we should look into more.  Overall, I feel that this boils down to localization, which is an ongoing discussion in #114.

**Comment by bershanskiy** [comment]
@vinyldarkscratch Let's split this PR into multiple. I dropped non-data changes to focus this commit on `api/` directory only. Is this PR small enough to review?

**Comment by bershanskiy** [comment]
FYI, all changes are basically one-liners. Some files have large diffs because the change is in the dictionary key and after sorting the order of features changes so diffs look rather large.

**Comment by vinyldarkscratch** [comment]
Thanks for splitting it up!  Yeah, this looks great, and we should be able to review it pretty soon.  We should also take the description from this and apply it to a PR that contains the linter script (don't worry if it fails without the data changes; that just shows it's working), and write up a new description -- one that doesn't have the "fixes" keyword.  Can we do that?

File: api/CSSMathNegate.json [hunk 1]
```diff
@@ -96,7 +96,7 @@
           }
         }
       },
-      "values": {
+      "value": {
```
Comment (vinyldarkscratch) [inline api/CSSMathNegate.json]: We should probably split off changes like this into another PR for scope-sake (since this is less the URL and more the feature identifier).  Can we do that?

File: api/DOMHighResTimestamp.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "api": {
-    "DOMHighResTimestamp": {
+    "DOMHighResTimeStamp": {
```
Comment (vinyldarkscratch) [inline api/DOMHighResTimestamp.json]: Same here!

File: api/Document.json [hunk 3]
```diff
@@ -9955,7 +9955,6 @@
       },
       "wasDiscarded": {
         "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/width",
```
Comment (vinyldarkscratch) [inline api/Document.json]: Looks like the URL was simply removed on this, rather than adjusted.  I looked and found there's no page for this one, which is something we should probably look into some more later on.  For now, we should probably keep an MDN URL around?

File: api/Headers.json [hunk 4]
```diff
@@ -113,9 +113,10 @@
           "deprecated": false
         }
       },
-      "append": {
+      "Headers": {
```
Comment (vinyldarkscratch) [inline api/Headers.json]: Ah, looks like the git diff butchered this one up -- it appears that you have potentially changed the feature identifiers, thus requiring an alteration to the sorting, correct?  We should probably separate this into its own PR for better review!

File: api/MediaKeyStatusMap.json [hunk 5]
```diff
@@ -239,9 +239,9 @@
           }
         }
       },
-      "iterator": {
+      "keys": {
```
Comment (vinyldarkscratch) [inline api/MediaKeyStatusMap.json]: Looks like we have another identifier change here.

File: api/Document.json (hunk 3 above)
Comment (bershanskiy) [reply api/Document.json]: > For now, we should probably keep an MDN URL around?

What value should it have? Pointing `wasDiscarded` to `width` seems pointless; should it be `Document`?

Comment (vinyldarkscratch) [reply api/Document.json]: I'd say the value should be `https://developer.mozilla.org/en-US/docs/Web/API/Document/wasDiscarded` -- even if the page doesn't exist right now.  (We've got a LOT of pages to create, apparently...)

Comment (bershanskiy) [reply api/Document.json]: For now, I made it `https://developer.mozilla.org/en-US/docs/Web/API/Document` so that people don't experience as many broken URLs.
